oauth2-example-app/src/test/java/com/clouway/oauth2/exampleapp/LoginEndpointTest.java

```diff
@@ -59,6 +61,7 @@ public void wrongAuthentication() throws Exception {
       will(Expectations.returnValue(remoteAddress));
       oneOf(request).getParameter("continue");
       will(returnValue("/xxx"));
+      allowing(request).getRequestURI();
```

Comment:
It's a bad idea to use allowing. This means that it can be called any number of times, and if you write 100 lines of request.getRequestURI the test will still pass. In the future if you wish to show that a method is called many times it would be better to specify how many times it is called with *exactly(n).of*. 
Here's a cheat sheet for jMock http://www.jmock.org/cheat-sheet.html

Reply:
Changed to OneOf